pydantic_ai_slim/pydantic_ai/exceptions.py

```diff
@@ -54,3 +54,17 @@ def __str__(self) -> str:
             return f'{self.message}, body:\n{self.body}'
         else:
             return self.message
+
+
+class UsageLimitExceeded(RuntimeError):
```

Comment:
should this inherit from `UnexpectedModelBehavior`?

Reply:
It can, but I didn’t think it really qualified as unexpected behavior, when we raise that elsewhere it’s more of a “something is fundamentally wrong”. But I can see the argument for considering this unexpected behavior, I just feel like you might want to handle unexpected behavior that _isn’t_ a limit exceeded, so maybe there should be a separate parent class of both. Happy to change this to just being a subclass of ModelUnexpectedBehavior if you prefer

Reply:
Common base class makes sense to me.